### PR TITLE
Created a local store for messages with route rules

### PR DESCRIPTION
**Pull request by BernardGatt**
Issue: https://github.com/customerio/issues/issues/9315

- Created a local store for messages with route rules that don't match the current route.
- On route change, we now check if there are any valid messages that should be displayed.

### CODE REVIEW

File: Gist/Managers/MessageQueueManager.swift [hunk 1]
```diff
@@ -18,6 +19,26 @@ class MessageQueueManager {
         }
     }
 
+    func checkLocalStoreForMessages() {
```
Comment (levibostian) [inline Gist/Managers/MessageQueueManager.swift]: Would this function benefit from a rename [such as in Android](https://github.com/customerio/gist-android/pull/2#discussion_r1098502444)?

**Review by levibostian — APPROVED** [review]

File: Gist/Managers/MessageQueueManager.swift [hunk 2]
```diff
@@ -3,6 +3,7 @@ import UIKit
 
 class MessageQueueManager {
     private var queueTimer: Timer!
+    private var localMessageStore: [String: Message] = [:]
```
Comment (levibostian) [inline Gist/Managers/MessageQueueManager.swift:7]: I think this code would benefit from a comment explaining what value this store provides. 

Explaining why this feature got added would be helpful for future reference.

**Review by levibostian — APPROVED** [review]

File: Gist/Managers/MessageQueueManager.swift (hunk 1 above)
Comment (BernardGatt) [reply Gist/Managers/MessageQueueManager.swift]: Updated both this and another internal method to match.

File: Gist/Managers/MessageQueueManager.swift (hunk 2 above)
Comment (BernardGatt) [reply Gist/Managers/MessageQueueManager.swift:7]: Added a comment explaining what the local store is used for

**Pull request merged**